domain/network/state/subnet.go

```diff
@@ -0,0 +1,447 @@
+
+	rows, err := tx.ExecContext(ctx, q, spaceID, uuid)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	affected, err := rows.RowsAffected()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if affected != 1 {
+		return errors.NotFoundf("subnet %q", uuid)
+	}
+
+	return nil
+}
+
+// UpdateSubnet updates the subnet identified by the passed uuid.
+func (st *State) UpdateSubnet(
+	ctx context.Context,
+	uuid string,
+	spaceID string,
+) error {
+	db, err := st.DB()
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	return db.StdTxn(ctx, func(ctx context.Context, tx *sql.Tx) error {
+		return updateSubnetSpaceIDTx(ctx, tx, uuid, spaceID)
+	})
+}
+
+// DeleteSubnet deletes the subnet identified by the passed uuid.
+func (st *State) DeleteSubnet(
+	ctx context.Context,
+	uuid string,
+) error {
+	db, err := st.DB()
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	deleteSubnetStmt := "DELETE FROM subnet WHERE uuid = ?;"
+	deleteSubjectSubnetAssociationStmt := "DELETE FROM subnet_association WHERE subject_subnet_uuid = ?;"
+	deleteAssociatedSubnetAssociationStmt := "DELETE FROM subnet_association WHERE associated_subnet_uuid = ?;"
+	selectProviderNetworkStmt := "SELECT provider_network_uuid FROM provider_network_subnet WHERE subnet_uuid = ?;"
+	deleteProviderNetworkStmt := "DELETE FROM provider_network WHERE uuid = ?;"
+	deleteProviderNetworkSubnetStmt := "DELETE FROM provider_network_subnet WHERE subnet_uuid = ?;"
+	deleteProviderSubnetStmt := "DELETE FROM provider_subnet WHERE subnet_uuid = ?;"
+	deleteAvailabilityZoneSubnetStmt := "DELETE FROM availability_zone_subnet WHERE subnet_uuid = ?;"
+
+	return db.StdTxn(ctx, func(ctx context.Context, tx *sql.Tx) error {
+		if _, err := tx.ExecContext(ctx, deleteSubjectSubnetAssociationStmt, uuid); err != nil {
+			return errors.Trace(err)
+		}
+		if _, err := tx.ExecContext(ctx, deleteAssociatedSubnetAssociationStmt, uuid); err != nil {
+			return errors.Trace(err)
+		}
+
+		row := tx.QueryRowContext(ctx, selectProviderNetworkStmt, uuid)
+		var providerNetworkUUID string
+		err = row.Scan(&providerNetworkUUID)
+		if err != nil {
+			return errors.Trace(err)
+		}
+
+		delProviderNetworkSubnetResult, err := tx.ExecContext(ctx, deleteProviderNetworkSubnetStmt, uuid)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		delProviderNetworkSubnetAffected, err := delProviderNetworkSubnetResult.RowsAffected()
+		if err != nil {
+			return errors.Trace(err)
+		}
+		if delProviderNetworkSubnetAffected != 1 {
+			return fmt.Errorf("provider network subnets for subnet %s not found", uuid)
+		}
+
+		delProviderNetworkResult, err := tx.ExecContext(ctx, deleteProviderNetworkStmt, providerNetworkUUID)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		delProviderNetworkAffected, err := delProviderNetworkResult.RowsAffected()
+		if err != nil {
+			return errors.Trace(err)
+		}
+		if delProviderNetworkAffected != 1 {
+			return fmt.Errorf("provider network for subnet %s not found", uuid)
+		}
```

Comment:
Compress these if statements...

```suggestion
		
		if affected, err := delProviderNetworkResult.RowsAffected(); err != nil {
			return errors.Trace(err)
		} else if affected != 1 {
			return fmt.Errorf("provider network for subnet %s not found", uuid)
		}
```

Reply:
done